tests/CMakeLists.txt

```diff
@@ -86,6 +87,10 @@ add_test_case(mqtt_connection_reconnection_backoff_unstable)
 add_test_case(mqtt_connection_reconnection_backoff_reset)
 add_test_case(mqtt_connection_reconnection_backoff_reset_after_disconnection)
 
+add_test_case(mqtt_validation_failure_publish_qos)
+add_test_case(mqtt_validation_failure_subscribe_empty)
+add_test_case(mqtt_validation_failure_unsubscribe_null)
```

Comment:
The added the unit test is testing for NULL topic, while we added the validation for NULL payload. Should we have a test for the null payload as well?

Reply:
Null payloads are fine so it's not really a validation test.  We could theoretically add a null payload test to standard client tests and verify it works as expected.  I'll add that when I get to client behavior.